### PR TITLE
Search user and charity

### PR DESCRIPTION
**Pull request by nurgazydev**
Search

### CODE REVIEW

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/SearchCharityResponse.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+public class SearchCharityResponse {
+
+    private Long charityId;
+    private SearchUserResponse saveUserResponse;
+    private String charityImage;
+    private String charityName;
+    private String charityCondition;
+    private LocalDate createdDate;
+    private UserFeedResponse reservoirUser;
+}
```
Comment (Sunatbekd) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/SearchCharityResponse.java]: Good job!

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/SearchCharityResponse.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@AllArgsConstructor
+@Builder
+public class SearchCharityResponse {
+
+    private Long charityId;
+    private SearchUserResponse saveUserResponse;
+    private String charityImage;
+    private String charityName;
+    private String charityCondition;
+    private LocalDate createdDate;
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/dto/responses/SearchCharityResponse.java]: createdAt

File: pom.xml [hunk 3]
```diff
@@ -153,8 +153,8 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>11</source>
-                    <target>11</target>
+                    <source>15</source>
```
Comment (i-Coderr) [inline pom.xml:156]: Why?

File: src/main/java/kg/peaksoft/giftlistb6/db/services/CharityService.java [hunk 4]
```diff
@@ -150,4 +153,43 @@ public SimpleResponse reserveCharity(Long charityId, boolean is) {
 
         return new SimpleResponse("Оk", "Бронирован!");
     }
+
+    public SearchAllResponse searchCharity(String text, String condition, String category, String subCategory) {
+        if (Objects.equals(condition, "Все")) {
+            condition = null;
+        } else if (Objects.equals(condition, "все")) {
```
Comment (i-Coderr) [inline src/main/java/kg/peaksoft/giftlistb6/db/services/CharityService.java:160]: Все

**Review by i-Coderr — COMMENTED** [review]
Commented at 8.58 pm

File: pom.xml (hunk 3 above)
Comment (nurgazydev) [reply pom.xml:156]: need it

Comment (nurgazydev) [reply pom.xml:156]: look at the search charity query

Comment (nurgazydev) [reply pom.xml:156]: @Query("""
           select c from Charity c
           where (lower(c.name) like lower(concat(:text,'%')) or :text is null)
           and (lower(c.condition) like lower(concat(:condition,'%')) or :condition is null)
           and (lower(c.category.name) like lower(concat(:category,'%')) or :category is null)
           and (lower(c.subCategory.name) like lower(concat(:subCategory,'%')) or :subCategory is null)
           and c.user.email not like concat(:email,'%')
           """)
    List<Charity> searchCharity(@Param("text") String text,
                                @Param("condition") String condition,
                                @Param("category") String category,
                                @Param("subCategory") String subCategory,
                                String email);
   it healps easily write query and make comfortable and does not affect the other

File: src/main/java/kg/peaksoft/giftlistb6/dto/responses/SearchCharityResponse.java (hunk 2 above)
Comment (nurgazydev) [reply src/main/java/kg/peaksoft/giftlistb6/dto/responses/SearchCharityResponse.java]: I will fix it

File: src/main/java/kg/peaksoft/giftlistb6/db/services/CharityService.java (hunk 4 above)
Comment (nurgazydev) [reply src/main/java/kg/peaksoft/giftlistb6/db/services/CharityService.java:160]: Des not depend on a capital letter or a small one

File: pom.xml (hunk 3 above)
Comment (i-Coderr) [reply pom.xml:156]: Ok

**Review by i-Coderr — APPROVED** [review]
Ok, let see

**Pull request merged**